swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/model/Resource.kt

```diff
@@ -119,6 +120,22 @@ interface Identifiable : Named {
   val resourceType: String
   val cloudProvider: String
   val grouping: Grouping?
+
+  /**
+   * Returns the opt-out url for this resource
+   * This is used for notifications
+   */
+  fun optOutUrl(workConfiguration: WorkConfiguration): String {
```

Comment:
Does this have any impact on json serialization/deserialization? aka will making changes here lead to problems reading from the current database when we roll it out?

Reply:
I dont believe so. Will test that out.